doc/doxygen/headers/glossary.h

```diff
@@ -584,8 +584,66 @@
  * template match a certain interface or behave in a certain way: such
  * constraints are called <em>concepts</em> in C++. See the discussion in
  * @ref Concepts for more information and a list of concepts in deal.II.
+ *
+ * (As a side note, concepts have been proposed as a language extension to
+ * C++ for a long time already. They would allow us to describe that a class
+ * or function has certain properties in order to be a qualified template
+ * argument. For example, it would allow us to express in C++ code that the
+ * first argument to, say, GridTools::find_closest_vertex(), must have a type
+ * that represents an actual mesh -- which we can currently only describe in
+ * words, see @ref ConceptMeshType "MeshType concept". Using C++ concepts
+ * would allow us to describe this in code and trying to call such a function
+ * with an object as first argument that is not, in fact, a mesh would yield
+ * a compiler error that makes the mismatch clear.
+ *
+ * Unfortunately, these proposals to C++ have never made it into the official
+ * C++ standards until C++20. We may start to use them once the vast majority
```

Comment:
I think this sentence is awkwardly phrased. I would write

Unfortunately, Concepts are not part of the current C++ standard and will be added, at the earliest, in the C++20 standard.